mockgcp/mockbigtable/cluster.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockbigtable
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	longrunning "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/pkg/storage"
+	pb "google.golang.org/genproto/googleapis/bigtable/admin/v2"
+	// pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/bigtable/admin/v2"
```

Comment:
Is this intended?  I think we want to mock the proto anyways to control the expectations.
Do you mind adding a comment about this change?

Reply:
The challenge is that Terraform uses GRPC today, and so if we change the name we might have problems.  I agree consistency would probably be better, so let me take a look at what we can do.

(The other option is to always use the "real" protos when we can - i.e. take this approach everywhere - the problem is that we might lose some control over the evolution of the proto schema.  But OTOH we aren't changing them anyway)

Reply:
So the issue is that we get "unknown service google.bigtable.admin.v2.BigtableInstanceAdmin", because we have renamed that service to `mockgcp.bigtable.admin.v2...` to avoid collisions.

Given this, I propose we go with what we have here.  If we can use the Protos as is, and only generate the GRPC gateway bindings, that _is_ easier (as we don't have to mess with the Protos).  It won't work for everything, but I proposed we use this one as an experiment for this simpler approach.

I'll add a comment!